### DISCUSSION TITLE
demo guidelines - allow starting later in an episode?

### DISCUSSION THREAD

**Discussion by ChristinaLK** [carpentries/trainers#87] [Trainer Practices]
Sparked by some of @Denubis comments in carpentries/instructor-training#1152, are there lesson episodes that would be good candidates for demo sessions but have long expository intros that prevent people from getting to the live coding quickly? Would it be possible (desirable) to have a list of episodes where we can use them for demos but recommend starting at a particular heading (after the introductory material)?

Example: [R Ecology, starting w/ data](https://datacarpentry.org/R-ecology-lesson/02-starting-with-data.html)

**Comment by brownsarahm** [comment]
I *really* like this idea. It will take some curating and some possibly reworking the table to make that information easy to read/display, but I think it will make demos better

**Comment by Denubis** [comment]
We should experiment, perhaps?

Thoughts:
1. How do we manage expectations of instructors-to-be and trainers?
2. How do we test to see how well episodes are introduced? (One of my things I look for is how well they set expectations and run a short intro, though this is very much secondary to live-coding)
3. I assume that we'll be skipping to the start of any code sections (and that we'll need to adjust our allowed/not-allowed modules accordingly) -- how do we frame this during the demo?

But yes, I agree with @brownsarahm that this could very much be a change for the better. We could also gain far more episodes if we made mandatory setup per lesson, but I think that that just adds to the prep load.

**Comment by Denubis** [comment]
@brownsarahm Do we want to run an experiment with this, somehow?

**Comment by brownsarahm** [comment]
I think so. Maybe we find a few trainers with upcoming Demos who want to try and ask them to report back here what they did and how it went? I think a few unstructured attempts makes sense to gather information on how to scale it up to a more formal pilot/trial period?  Unless @karenword sees a particular reason not to, or she or @klbarnes20 has a better idea on how to experiment. (Kelly could potentially provide veto rationale, but I expect Karen's longer tenure is more likely to provide such an example).

**Comment by Denubis** [comment]
Just writing this out as a proposal in one comment box:

1. We propose to experiment with instructing new instructors to start partially through an episode as part of their checkout, so that we can make sure they demonstrate live coding.
2. @brownsarahm and I need to work on text-scripts that we will put on the checkout page, the rubric, emails-to-instructors, and the trainer running the demo page.
3. We need to figure out what lessons should be part of this initial set, so that we're not reviewing *all* possible lessons.

Any other thoughts as part of prep?

**Comment by karenword** [comment]
I think @brownsarahm suggestion of tossing this out for an informal pilot is a good one (not so much an experiment, really). The main thing is not to create undue burden on the hosting Trainer, I think, in trying to decide or communicate about where to start. Especially with a full demo, it's important that the "where to start" communication be swift and easy. If it sounds like this is manageable, we can proceed to think about step 2 in @Denubis' helpful list. If it sounds like there are hassles, we could jump to step 3 and see if we can make it easier for one or more lessons, then pilot those before step 2.

**Comment by Denubis** [comment]
So I apparently have a teaching demo next week. With open refine and a TBC (and some normal stuff). Do we think it's worth getting scripts prepared for that demo? I'd need to send out notifications soon if we're doing things in an experimental fashion. Otherwise, I can put down for a september demo and advertise it as experimental?

I'm going to email them now to get consent to even explore this with them.

**Comment by brownsarahm** [comment]
What do you mean by scripts? What you as trainer host will say to describe what's about to happen?  

Also, if they're preparing as instructed, how much do the trainees need to be notifified things may be "different" they've never experienced a "regular" teaching demo so it's novel either way.

**Comment by Denubis** [comment]
By scripts, I mean the text blocks like the ones on https://docs.carpentries.org/topic_folders/instructor_training/email_templates_trainers.html and the stuff I prepare to open/close a session. While it's novel, we need to make sure it's not contradicting any of the sources of instruction (rubrics, training, etc) that set expectations.
